### PR TITLE
bridge onramp

### PR DESCRIPTION
**Pull request by dcposch**
## Checklist

- [x] Connect Wallet + fetch available assets
- [x] Integrate Squid Router
- [ ] Testing & reliability

## 🙂

Screenshot 2023-11-10 at 04 23 51

## 😱

Screenshot 2023-11-10 at 04 23 16

guess i'll test tomorrow

### CODE REVIEW

File: packages/daimo-api/src/api/bridge.ts [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+      symbol: "USDC.e",
+      addr: "0x2791Bca1f2de4661ED88A30C99A7a9449Aa84174",
+      decimals: 6,
+    },
+    {
+      chainSlug: "polygon",
+      symbol: "DAI",
+      addr: "0x8f3Cf7ad23Cd3CaDbD9735AFf958023239c6A063",
+      decimals: 18,
+    },
```
Comment (nalinbhardwaj) [inline packages/daimo-api/src/api/bridge.ts:95]: no ETH/WETH? most people should have that right

File: apps/daimo-web/src/app/bridge/page.tsx [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+  useEffect(() => {
+    setRoute(undefined);
+    (async () => {
+      const params = {
+        fromChain: fromBal.chainID,
+        fromToken: fromBal.tokenAddr,
+        fromAddress: fromBal.ownerAddr,
+        fromAmount: amount,
+        toChain: 8453, // Base
+        toToken: "0x833589fCD6eDb6E08f4c7C32D4f71b54bdA02913", // USDC
```
Comment (nalinbhardwaj) [inline apps/daimo-web/src/app/bridge/page.tsx:222]: nit use tokenMetadata from common

File: apps/daimo-web/src/app/bridge/page.tsx [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+
+function RouteSummary({ route }: { route: RouteData }) {
+  const timeS = route.estimate.estimatedRouteDuration;
+  const amount = (Number(route.estimate.toAmount) / 1e6).toFixed(2);
+
+  return (
+    <div>
+      <div className="flex justify-between items-center pl-2 pr-8 py-4">
+        <div className="flex items-center">
+          <AccountBubble name="dcposch" />
```
Comment (nalinbhardwaj) [inline apps/daimo-web/src/app/bridge/page.tsx:292]: flagging todo

File: packages/daimo-api/src/router.ts [hunk 4]
```diff
@@ -48,12 +49,23 @@ export function createRouter(
...
+    .use(corsMiddleware)
+    .use(tracerMiddleware);
 
   return trpcT.router({
     health: publicProcedure.query(async (_opts) => {
-      return "healthy";
+      // Push Notifier is the last service to load.
+      const ret = notifier.isInitialized ? "healthy" : "pending";
+      console.log(`[API] health check: ${ret}`);
+      return ret;
```
Comment (nalinbhardwaj) [inline packages/daimo-api/src/router.ts:66]: i think the pending still returns 200OK, you should throw an error for #388

File: packages/daimo-api/src/api/bridge.ts (hunk 1 above)
Comment (dcposch) [reply packages/daimo-api/src/api/bridge.ts:95]: v2

File: apps/daimo-web/src/app/bridge/page.tsx (hunk 3 above)
Comment (dcposch) [reply apps/daimo-web/src/app/bridge/page.tsx]: done

**Comment by dcposch** [comment]
## Squid is too expensive

Costs ~$10 to bridge from Optimism to Base (??)

Switching to li.fi in follow-up PR

**Pull request closed without merging**